internal/sasl/challenge.go

```diff
@@ -65,3 +65,16 @@ func ParseChallenge(challenge []byte) (*Challenge, error) {
 
 	return &ch, nil
 }
+
+// HighestQopLevel extracts most secure Qop level from the list provided as an argument.
+func HighestQopLevel(qopList []string) string {
```

Comment:
Instead of this function, maybe change the returned challenge to have sorted Qops?

Reply:
Hi! Thanks for this great implementation!
I am in exactly the same configuration (HDFS 3.1.1, HDP 3.1.4.0) but have to run on other clusters as well and would love to have this modification merged upstream.
Could you please explain how you would want this to be implemented? I'd happily try to contribute a patch.
BR